bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -198,11 +208,16 @@ private State getValue(String channelId, GoEStatusResponseDTO goeResponse) {
                     count++;
                 }
                 return new DecimalType(count);
-            case TEMPERATURE:
-                if (goeResponse.temperature == null) {
+            case TEMP_TMA1:
+                if (goeResponse.temptma == null) {
                     return UnDefType.UNDEF;
                 }
-                return new QuantityType<>(goeResponse.temperature, SIUnits.CELSIUS);
+                return new QuantityType<>(goeResponse.temptma[0], SIUnits.CELSIUS);
```

Comment:
You could add a check if this array is at least of the size 1. If the array is empty, this binding would crash.

Reply:
I have to find out how to do that, this is my first submission to github.

Reply:
You can check the size of the array beforehand with `if(goeResponse.temptma.length >= 1)`. Don't hesitate to ask if something is unclear. You can keep this pull request open for the time you're fixing the findings.